relay/relay.go

```diff
@@ -529,135 +416,13 @@ func (r *Relay) process(parentCtx context.Context) error {
 			if err != nil {
-				return errors.Trace(err)
+				return errors.Annotatef(err, "save position %s, GTID sets %v into meta", lastPos, lastGTID)
 			}
 		}
 	}
 }
 
-// addFlagToEvent adds flag to binlog event
-func (r *Relay) addFlagToEvent(e *replication.BinlogEvent, f uint16, eventFormat *replication.FormatDescriptionEvent) {
-	newF := e.Header.Flags | f
-	// header structure:
-	// 4 byte timestamp
-	// 1 byte event
-	// 4 byte server-id
-	// 4 byte event size
-	// 4 byte log pos
-	// 2 byte flags
-	startIdx := 4 + 1 + 4 + 4 + 4
-	binary.LittleEndian.PutUint16(e.RawData[startIdx:startIdx+2], newF)
-	e.Header.Flags = newF
-
-	// re-calculate checksum if needed
-	if eventFormat == nil || eventFormat.ChecksumAlgorithm != replication.BINLOG_CHECKSUM_ALG_CRC32 {
-		return
-	}
-	calculatedPart := e.RawData[0 : len(e.RawData)-replication.BinlogChecksumLength]
-	checksum := crc32.ChecksumIEEE(calculatedPart)
-	binary.LittleEndian.PutUint32(e.RawData[len(e.RawData)-replication.BinlogChecksumLength:], checksum)
-}
-
-// detectGap detects whether gap exists in relay log file
-func (r *Relay) detectGap(e *replication.BinlogEvent) (bool, uint32, error) {
-	if r.fd == nil {
-		return false, 0, nil
-	}
-	fi, err := r.fd.Stat()
-	if err != nil {
-		return false, 0, errors.Trace(err)
-	}
-
-	size := uint32(fi.Size())
-	if e.Header.LogPos-e.Header.EventSize > size {
-		return true, size, nil
-	}
-	return false, size, nil
-}
-
-// compareEventWithFileSize compares event's start pos with relay log file size
-// returns result:
-//   -1: less than file size
-//    0: equal file size
-//    1: greater than file size
-func (r *Relay) compareEventWithFileSize(e *replication.BinlogEvent) (result int, fSize uint32, err error) {
-	if r.fd != nil {
-		fi, err := r.fd.Stat()
-		if err != nil {
-			return 0, 0, errors.Annotatef(err, "compare relay log file size with %+v", e.Header)
-		}
-		fSize = uint32(fi.Size())
-		startPos := e.Header.LogPos - e.Header.EventSize
-		if startPos < fSize {
-			return -1, fSize, nil
-		} else if startPos > fSize {
-			return 1, fSize, nil
-		}
-	}
-	return 0, fSize, nil
-}
-
-// handleFormatDescriptionEvent tries to create new binlog file and write binlog header
-func (r *Relay) handleFormatDescriptionEvent(filename string) (exist bool, err error) {
-	if r.fd != nil {
-		// close the previous binlog log
-		r.fd.Close()
-		r.fd = nil
-	}
-
-	if len(filename) == 0 {
-		binlogReadErrorCounter.Inc()
-		return false, errors.NotValidf("write FormatDescriptionEvent with empty binlog filename")
-	}
-
-	fullPath := path.Join(r.meta.Dir(), filename)
-	fd, err := os.OpenFile(fullPath, os.O_CREATE|os.O_RDWR, 0644)
-	if err != nil {
-		return false, errors.Annotatef(err, "file full path %s", fullPath)
-	}
-	r.fd = fd
-
-	// record current active relay log file, and keep it until newer file opened
-	// when current file's fd closed, we should not reset this, because it may re-open again
-	r.setActiveRelayLog(filename)
```

Comment:
do we need to call `setActiveRelayLog` somewhere?

Reply:
Good catch! add a call to `setActiveRelayLog ` in https://github.com/pingcap/dm/pull/171/commits/f9b1663d8a755052f0307248dd1f9a62ead84dc5.